spot-client/src/common/remote-control/xmpp-connection.js

```diff
@@ -314,6 +314,15 @@ export default class XmppConnection {
         return this._isXmppConnectionActive;
     }
 
+    /**
+     * Returns whether or not the internal XMPP connection is being reconnected.
+     *
+     * @returns {boolean}
+     */
+    isReconnecting() {
+        return Boolean(this._silentReconnectPromise) || Boolean(this.initPromise);
```

Comment:
Im thinking that `this.initPromise` should be reset after the join Muc process is done then I could re-use it for that purpose.

On a side note it doesn't seem like a correct behavior to return this promise because the options could be different and in general joinMuc should not be called without leaving the Muc first.

Reply:
attempt to address in: https://github.com/jitsi/jitsi-meet-spot/pull/681/commits/16a906bdfb43f007e76d80e3d5e6a1aa84f5c4ec